sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -551,12 +551,30 @@ class annotated_ref {
     operator T() const;
     T operator=(T) const;
     T operator=(const annotated_ref&) const;
+
     // OP is: +=, -=, *=, /=, %=, <<=, >>=, &=, |=, ^=
-    T operatorOP(T) const;
+    // is_ann_ref_v<O> returns true if O is a annotated_ref<T2,P2> type
+    // where T2 and P2 can be any annotated_ref supported types
+    template <typename O, typename = std::enable_if_t<!is_ann_ref_v<O>>>
```

Comment:
Our convention is that for the spec we don't put the enable_if and instead write:
`template <typename O> //available only if O is not an annotated_ref`

(copied comment into a new conversation because I accidentally created two separate things int same)